### PR TITLE
Extend sign in form with username

### LINKED ISSUE #43: Allow to use username as login
Need to allow use username as login, but if string contains @ need to validate as email address

### PR DESCRIPTION
**Pull request by stenvix**
Close #43

### CODE REVIEW

File: lib/configs/app_constans.dart [hunk 1]
```diff
@@ -27,6 +27,7 @@ class AppConstants {
   static String tripsSearchToDate = 'To date';
 
   static String emailInputLabel = 'Enter email';
+  static String loginInputLabel = 'Enter email or username';
```
Comment (taruuuch) [inline lib/configs/app_constans.dart:30]: Change label form 'Enter email' to 'mail@example.com' and 'mail@example.com or username'

File: lib/services/form_validators.dart [hunk 2]
```diff
@@ -1,34 +1,51 @@
 class FormValidators {
   static String emailRegex = r"^[a-zA-Z0-9.]+@[a-zA-Z0-9]+\.[a-zA-Z]+";
+  static String loginRegex = r"^[a-z0-9_.-]*$";
   static var _validatorConstants = {
-    'empty_email'    : 'Enter email',
-    'invalid_email'  : 'Invalid email',
-    'emply_password' : 'Enter password',
+    'empty_email': 'Enter email',
+    'empty_login': 'Enter email or username',
+    'emply_password': 'Enter password',
```
Comment (taruuuch) [inline lib/services/form_validators.dart]: Add to all constants 'Please enter...'

File: lib/services/form_validators.dart [hunk 3]
```diff
@@ -1,34 +1,51 @@
...
+    'invalid_email': 'Invalid email',
+    'invalid_username': 'Invalid username',
```
Comment (taruuuch) [inline lib/services/form_validators.dart]: Add to all constants 'Please enter a valid ...'

**Review by taruuuch — COMMENTED** [review]
_firstNameLabel_ change to 'e.g: John'
_lastNameLabel_ change to 'e.g: Smith'

File: lib/configs/app_constans.dart (hunk 1 above)
Comment (stenvix) [reply lib/configs/app_constans.dart:30]: Placeholder and label are different things, which one I need to change?

File: lib/services/form_validators.dart (hunk 2 above)
Comment (stenvix) [reply lib/services/form_validators.dart]: Done

File: lib/services/form_validators.dart (hunk 3 above)
Comment (stenvix) [reply lib/services/form_validators.dart]: Done

File: lib/configs/app_constans.dart (hunk 1 above)
Comment (stenvix) [reply lib/configs/app_constans.dart:30]: Moved to #46

**Pull request merged**